lib/events/athena/consumer.go

```diff
@@ -0,0 +1,457 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package athena
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"io"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3Types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	sqsTypes "github.com/aws/aws-sdk-go-v2/service/sqs/types"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	// maxWaitTimeOnReceiveMessageFromSQS defines how long single
+	// receiveFromQueue will wait if there is no max events (10).
+	maxWaitTimeOnReceiveMessageFromSQS = 5 * time.Second
+	// maxNumberOfWorkers defines how many workers are processing messages
+	// from queue or writing parquet files to s3.
+	maxNumberOfWorkers = 5
```

Comment:
Why was the number 5 chosen?

Reply:
I have hardcode 5 for now just based on gut feeling. In future probably this number should depend on how many items are in queue. 5 workers on my dev machine were enough to handle max load defined in cloud RFD (250 events/s if I remember correctly).